### PR TITLE
Bn 1201 minting validation

### PR DESCRIPTION
**Pull request by nandotorterolo**
## Purpose

Implement **case 1 AND case2**

```
Validations only for minting (After projection to min (only if for all inputs and outputs isMint == true))

Case 1: Group

policy is attached to transaction

at least 1 group token is minted

reference in policy contains LVLs (> 0)

Case 2: Series

policy is attached to transaction

at least 1 series token is minted

reference in policy contains LVLs (> 0)
```

Sequence of merge:

- BN-1201-well-formedness-b-no-repeated-utx-os <- BN-1201-mintingValidation
- Validations_Tam_v2  <- BN-1201-well-formedness-b-no-repeated-utx-os
- main <- Validations_Tam_v2

https://topl.atlassian.net/wiki/spaces/LABS/pages/678658061/Validation+TAMv2+Transactions+-+Meeting+Fernando+Edmundo+2023-09-08

## Testing
unit test and preparePR

## Tickets
https://topl.atlassian.net/browse/BN-1201

### CODE REVIEW

File: brambl-sdk/src/main/scala/co/topl/brambl/validation/TransactionSyntaxInterpreter.scala [hunk 1]
```diff
@@ -323,4 +325,84 @@ object TransactionSyntaxInterpreter {
...
+        .map(_.value.getGroup.groupId)
+
+    val seriesIdsOnMintedStatements =
+      transaction.inputs
+        .filter(_.value.value.isSeries)
+        .filter(sto => transaction.mintingStatements.map(_.seriesTokenUtxo).contains(sto.address))
+        .map(_.value.getSeries.seriesId)
+
+    transaction.outputs.filter { utxo =>
+      !utxo.value.value.isLvl ||
```
Comment (mundacho) [inline brambl-sdk/src/main/scala/co/topl/brambl/validation/TransactionSyntaxInterpreter.scala:350]: should be an AND

File: brambl-sdk/src/main/scala/co/topl/brambl/validation/TransactionSyntaxInterpreter.scala [hunk 2]
```diff
@@ -323,4 +325,84 @@ object TransactionSyntaxInterpreter {
...
+      !utxo.value.value.isTopl ||
```
Comment (mundacho) [inline brambl-sdk/src/main/scala/co/topl/brambl/validation/TransactionSyntaxInterpreter.scala:351]: should be an AND

**Review by mundacho — CHANGES_REQUESTED** [review]
There are some minor changes, the problem is not the code itself, but I think I found mistakes in the logic (some ors need to become AND)

**Review by DiademShoukralla — APPROVED** [review]
LGTM

**Pull request merged**